src/main/java/appeng/me/helpers/AENetworkProxy.java

```diff
@@ -325,14 +325,6 @@ public IEnergyGrid getEnergy() throws GridAccessException {
 
     @Nonnull
     private <T extends IGridCache> T getGridCache(Class<T> clazz) throws GridAccessException {
-        final IGrid grid = this.getGrid();
-        if (grid == null) {
-            throw new GridAccessException();
-        }
-        final T eg = grid.getCache(clazz);
-        if (eg == null) {
-            throw new GridAccessException();
```

Comment:
`GridAccessException` are actually often catched. In this case we would replace it with a NPE, which is usually not caught. It probably will not cause them, but the grid construction is not trivial, so it can run into edge cases. Also we should probably change the `Grid` constructor into a factory, so it no longer leaks the object through the event bus before fully initialized. 
This should eliminate the need to check as the caches a populated during the constructor, assuming nothing breaks it and it suddenly queries a non existing cache.

Reply:
No, since if you trace this value back, it CANNOT be null at this point since it was already de-referenced in getCache